src/simuq/dwave/dwave_sim_provider.py

```diff
@@ -0,0 +1,132 @@
+from simuq.provider import BaseProvider
+from simuq.solver import generate_as
+import numpy as np
+import json
+import re
+from neal import SimulatedAnnealingSampler
+
+from simuq.aais import ising
+from simuq.dwave.dwave_transpiler import DwaveTranspiler
+
+
+class DWaveSimProvider(BaseProvider):
+    def __init__(self, **sampler_kwargs):
+        # Initialize without API key requirements
+        super().__init__()
+        self._samples = None
+        # Store all kwargs for sampling methods
+        self.sampling_args = sampler_kwargs
+
+    def compile(self,
+                qs,
+                anneal_schedule=[[0, 0], [20, 1]],
```

Comment:
The annealing schedule does not correspond to simulated annealing. There are other parameters such as the beta_range, beta_schedule_type, num_sweeps, etc. See if you can implement it maybe as kwargs. Details here https://github.com/dwavesystems/dwave-neal